### ISSUE TITLE
Fix false positives for variables in selector-type-case

### ISSUE THREAD

**Issue by setek** [stylelint/stylelint#3391]
> What is the problem?

`selector-type-case` has no customisation settings, and so fails with styled-components’ `$dummyValue`

eg.

`✖  Expected "$dummyValue" to be "$dummyvalue"                        selector-type-case`

> What solution would you like to see?

an ignore setting akin to `selector-type-no-unknown`: `'selector-type-no-unknown': [true, {ignoreTypes: ['$dummyValue']}],`

eg.

`'selector-type-case': ['lower', {ignoreTypes: ['/dummyValue/']}],`

Thanks!

**Comment by jeddy3** [comment]
@sekek Thanks for the request and for using the template.

I think the rule should ignore variables by default as variables are not _type selectors_.

I assume this hasn't come up before as variables are, unlike in styled-components, seldom used within selectors in preprocessors like Sass and Less.

As the rule already makes use of the [`isStandardSyntaxTypeSelector` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxTypeSelector.js), it should just be the case of:

1. Adding a condition to [return early](https://github.com/stylelint/stylelint/blob/282121b0aa74feab577b2bf286be6e47b4c9227d/lib/utils/isStandardSyntaxValue.js#L17-L19) for variables to that util.
2. Adding a "This rule ignores variables." to the rule description.

This will also fix the behaviour of `selector-type-no-unknown`.

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by XhmikosR** [comment]
I'm hitting this too...

> 41:24  ×  Expected "The" to be "the"   selector-type-case

```less
.alert-dismissable, // The misspelled .alert-dismissable was deprecated in 3.2.0.
.alert-dismissible {
  padding-right: (@alert-padding + 20);

  // Adjust close link position
  .close {
    position: relative;
    top: -2px;
    right: -21px;
    color: inherit;
  }
}
```

**Comment by jeddy3** [comment]
@XhmikosR That looks like a different issue, perhaps related to comments and the Less parser. Can you please create a new issue, following the "Report a bug" template, so that we can better help resolve the issue for you.

**Comment by niksajanjic** [comment]
I'm using these 2 rules to fix the problem:

```js
{
  rules: {
    'selector-type-case': [true, {
      ignoreTypes: ['/^\\$\\w+/'],
    }],
    'selector-type-no-unknown': [true, {
      ignoreTypes: ['/^\\$\\w+/'],
    }],
  }
}
```

This ignores variables starting with `$`.

**Comment by Sowed** [comment]
I came here battling `stylelint` errors with **styled-components** custom elements. Thank you @niksajanjic , your solution was super helpful except for the `selector-type-case` option. It kept throwing the following warning with `stylelint: ^13.2.0` and linting failed for those rules.

> Invalid Option: Invalid option value "true" for rule "selector-type-case"

Changing the [option to](https://stylelint.io/user-guide/rules/selector-type-case) either **lower** or **upper** fixes the warning. This is my modified version of your rule.

```js

    "selector-type-case": [
      "lower",
      {
        "ignoreTypes": [
          "/^\\$\\w+/"
        ]
      }
    ],
    "selector-type-no-unknown": [
      true,
      {
        "ignoreTypes": [
          "/^\\$\\w+/"
        ]
      }
    ]
```
